src/script_interface/analysis/Analysis.cpp

```diff
@@ -81,6 +82,15 @@ Variant Analysis::do_call_method(std::string const &name,
     auto const local = particle_short_range_energy_contribution(pid);
     return mpi_reduce_sum(context()->get_comm(), local);
   }
+  if (name == "particle_neighbor_pids") {
+    auto const neighbor_pids = get_neighbor_pids();
+    std::unordered_map<int, std::vector<int>> dict;
+    std::for_each(neighbor_pids.begin(), neighbor_pids.end(),
+                  [&dict](NeighborPIDs const &neighbor_pid) {
+                    dict[neighbor_pid.pid] = neighbor_pid.neighbor_pids;
+                  });
```

Comment:
```suggestion
    std::unordered_map<int, std::vector<int>> dict;
    context()->parallel_try_catch([&]() {
      auto const neighbor_pids = get_neighbor_pids();
      std::for_each(neighbor_pids.begin(), neighbor_pids.end(),
                    [&dict](NeighborPIDs const &neighbor_pid) {
                      dict[neighbor_pid.pid] = neighbor_pid.neighbor_pids;
                    });
    });
```
Because the core function can throw an exception, we need a parallel try catch block.